tests/unit/test_idle_loop.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2024 Canonical Ltd.
+# Licensed under the Apache V2, see LICENCE file for details.
+from __future__ import annotations
+
+from typing import AbstractSet, Iterable
+
+from freezegun import freeze_time
```

Comment:
What do you think about making `now` an argument to `Loop.next` to eliminate this dependency? The nice `unroll` function you have here could just take an iterable of times as well as statuses for minimal refactoring of tests, which would (imo) be nicer than using the various freeze_gun commands. Not a blocker tho by any means

Reply:
I'd rather not.

Reply:
Fair enough, I don't think its necessary at all, just an idea